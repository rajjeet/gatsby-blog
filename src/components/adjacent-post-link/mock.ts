export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
