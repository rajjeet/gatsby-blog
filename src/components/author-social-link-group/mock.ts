import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
});

