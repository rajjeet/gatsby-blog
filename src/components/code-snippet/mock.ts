import { TProps } from './types';

export const makeProps = (): TProps => ({
  children: `
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
`,
  fileName: 'index.js',
});
