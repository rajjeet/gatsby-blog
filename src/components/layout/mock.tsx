import React from 'react';
import { TProps } from './types';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { siteMetadata } from '../../../gatsby-config';

export const makeProps = (): TProps => ({
  children: <div>Sample child</div>,
  data: {
    file: createMockGatsbyImageSharpFluid.file,
    site: siteMetadata,
  },
});

