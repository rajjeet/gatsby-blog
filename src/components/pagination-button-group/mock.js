export const makeProps = () => ({
  className: {}, currentPage: 1, numOfPages: 5, paginationSlug: '',
});
