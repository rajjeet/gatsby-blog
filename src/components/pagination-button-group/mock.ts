export const makeProps = ({ currentPage = 1, numOfPages = 5 } = {}): TProps => ({
  className: '', currentPage, numOfPages, paginationSlug: '',
});

export type TProps = {
  className: string;
  currentPage: number;
  numOfPages: number;
  paginationSlug: string;
};
