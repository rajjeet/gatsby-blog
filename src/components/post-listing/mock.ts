import { TProps } from './types';
import { mockPosts } from '../../fixtures/posts';

export const makeProps = ({
  posts = mockPosts,
  heading = 'Post Listing',
  numOfPages = 5,
  currentPage = 1,
  paginationSlug = '/sample_slug/',
} = {}): TProps => ({
  posts,
  heading,
  numOfPages,
  currentPage,
  paginationSlug,
});

