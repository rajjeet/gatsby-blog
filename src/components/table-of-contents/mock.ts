import { TProps } from './types';

export const makeProps = (): TProps => ({
  items: [
    {
      url: '#introduction',
      title: 'Introduction',
    },
    {
      url: '#the-basics',
      title: 'The Basics',
      items: [
        {
          url: '#state-management-redux',
          title: 'State Management (Redux)',
        },
        {
          url: '#context-api',
          title: 'Context API',
        },
      ],
    },
    {
      url: '#manage-global-state-in-react-hooks',
      title: 'Manage Global State in React Hooks',
    },
    {
      url: '#middleware-logger',
      title: 'Middleware: Logger',
    },
  ],
});

