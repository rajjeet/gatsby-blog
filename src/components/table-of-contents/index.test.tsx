import React from 'react';
import {
  render, RenderResult,
} from '@testing-library/react';
import { TableOfContents } from './index';
import { makeProps } from './mock';

function renderTableOfContents(): RenderResult {
  return render(<TableOfContents {...makeProps()} />);
}

describe('<TableOfContents />', () => {
  it('should render', () => {
    const { asFragment } = renderTableOfContents();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should show each link', () => {
    const { getByText } = renderTableOfContents();
    expect(getByText('Introduction')).toBeDefined();
    expect(getByText('The Basics')).toBeDefined();
    expect(getByText('State Management (Redux)')).toBeDefined();
    expect(getByText('Context API')).toBeDefined();
    expect(getByText('Manage Global State in React Hooks')).toBeDefined();
    expect(getByText('Middleware: Logger')).toBeDefined();
  });

  it('should show only one heading', () => {
    const { getByText } = renderTableOfContents();
    expect(getByText('Outline')).toBeDefined();
  });

  it('should show a "Back to Top" button', () => {
    const { getByText } = renderTableOfContents();
    expect(getByText('Top')).toBeDefined();
  });
});
