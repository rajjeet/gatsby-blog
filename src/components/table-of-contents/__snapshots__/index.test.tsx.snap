// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableOfContents /> should render 1`] = `
<DocumentFragment>
  <div
    class="sc-bwzfXH jFCYEt"
  >
    <h4
      class="sc-htpNat hHtpcc"
    >
      Outline
    </h4>
    <a
      class="sc-ifAKCX dutTtV"
      href="#"
    >
      <span>
        Top
      </span>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-alt-circle-up fa-w-16 "
        data-icon="arrow-alt-circle-up"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 256C8 119 119 8 256 8s248 111 248 248-111 248-248 248S8 393 8 256zm292 116V256h70.9c10.7 0 16.1-13 8.5-20.5L264.5 121.2c-4.7-4.7-12.2-4.7-16.9 0l-115 114.3c-7.6 7.6-2.2 20.5 8.5 20.5H212v116c0 6.6 5.4 12 12 12h64c6.6 0 12-5.4 12-12z"
          fill="currentColor"
        />
      </svg>
    </a>
  </div>
  <ul
    class="sc-bdVaJa cSwstg"
    id="static-toc"
  >
    <div>
      <a
        class="sc-bxivhb lmzons"
        href="#introduction"
      >
        <li>
          Introduction
        </li>
      </a>
    </div>
    <div>
      <a
        class="sc-bxivhb lmzons"
        href="#the-basics"
      >
        <li>
          The Basics
        </li>
      </a>
      <ul
        class="sc-bdVaJa cSwstg"
        id="static-toc"
      >
        <div>
          <a
            class="sc-bxivhb lmzons"
            href="#state-management-redux"
          >
            <li>
              State Management (Redux)
            </li>
          </a>
        </div>
        <div>
          <a
            class="sc-bxivhb lmzons"
            href="#context-api"
          >
            <li>
              Context API
            </li>
          </a>
        </div>
      </ul>
    </div>
    <div>
      <a
        class="sc-bxivhb lmzons"
        href="#manage-global-state-in-react-hooks"
      >
        <li>
          Manage Global State in React Hooks
        </li>
      </a>
    </div>
    <div>
      <a
        class="sc-bxivhb lmzons"
        href="#middleware-logger"
      >
        <li>
          Middleware: Logger
        </li>
      </a>
    </div>
  </ul>
</DocumentFragment>
`;
